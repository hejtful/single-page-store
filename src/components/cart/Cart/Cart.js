import React from 'react';

import { CartItem } from '../CartItem';
import { CartTotal } from '../CartTotal';
import { getTotalAmount } from 'utility/cart/getTotalAmount';

import styles from './Cart.module.css';

export const Cart = ({ items, onCartItemRemove }) => (
  <div className={styles.cart}>
    {items?.length ? (
      <>
        {items.map((item) => (
          <CartItem {...item} key={item.id} onRemove={onCartItemRemove} />
        ))}

        <CartTotal amount={getTotalAmount(items)} />
      </>
    ) : (
      <div className={styles.emptyWrapper}>Your cart is currently empty.</div>
    )}
  </div>
);
