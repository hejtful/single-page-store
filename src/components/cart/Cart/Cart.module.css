.cart {
}

.emptyWrapper {
  margin-top: 1em;
}
