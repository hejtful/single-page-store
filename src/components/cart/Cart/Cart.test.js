import React from 'react';
import { render } from '@testing-library/react';

import { Cart } from './Cart';

const initialCartItem = {
  id: '1',
  title: 'Test product title',
  price: 0.25,
  quantity: 1,
};

const additionalCartItem = {
  id: '2',
  title: 'Additional item title',
  price: 1000,
  quantity: 5,
};

const methods = {
  onCartItemRemove: jest.fn(),
};

it('renders a message indicating that the cart is empty if items prop is undefined', () => {
  const props = {};

  const { container } = render(<Cart {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('renders a message indicating that the cart is empty if items prop is an empty array', () => {
  const props = {
    items: [],
  };

  const { container } = render(<Cart {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('renders a single product', () => {
  const props = {
    items: [initialCartItem],
  };

  const { container } = render(<Cart {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('renders multiple products', () => {
  const props = {
    items: [initialCartItem, additionalCartItem],
  };

  const { container } = render(<Cart {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});
