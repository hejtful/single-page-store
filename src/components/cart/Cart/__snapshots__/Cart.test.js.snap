// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a message indicating that the cart is empty if items prop is an empty array 1`] = `
<div>
  <div
    class="cart"
  >
    <div
      class="emptyWrapper"
    >
      Your cart is currently empty.
    </div>
  </div>
</div>
`;

exports[`renders a message indicating that the cart is empty if items prop is undefined 1`] = `
<div>
  <div
    class="cart"
  >
    <div
      class="emptyWrapper"
    >
      Your cart is currently empty.
    </div>
  </div>
</div>
`;

exports[`renders a single product 1`] = `
<div>
  <div
    class="cart"
  >
    <div
      class="wrapper"
    >
      <div
        class="infoWrapper"
      >
        <div
          class="line"
        >
          <h4
            class="title"
          >
            Test product title
          </h4>
          <span>
            0.25
          </span>
        </div>
        <div
          class="line"
        >
          <span>
            Quantity:
          </span>
          <span>
            1
          </span>
        </div>
      </div>
      <div
        class="removeWrapper"
      >
        <button
          type="button"
        >
          X
        </button>
      </div>
    </div>
    <div
      class="wrapper"
    >
      Total: 
      0.25
      $
    </div>
  </div>
</div>
`;

exports[`renders multiple products 1`] = `
<div>
  <div
    class="cart"
  >
    <div
      class="wrapper"
    >
      <div
        class="infoWrapper"
      >
        <div
          class="line"
        >
          <h4
            class="title"
          >
            Test product title
          </h4>
          <span>
            0.25
          </span>
        </div>
        <div
          class="line"
        >
          <span>
            Quantity:
          </span>
          <span>
            1
          </span>
        </div>
      </div>
      <div
        class="removeWrapper"
      >
        <button
          type="button"
        >
          X
        </button>
      </div>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="infoWrapper"
      >
        <div
          class="line"
        >
          <h4
            class="title"
          >
            Additional item title
          </h4>
          <span>
            1000
          </span>
        </div>
        <div
          class="line"
        >
          <span>
            Quantity:
          </span>
          <span>
            5
          </span>
        </div>
      </div>
      <div
        class="removeWrapper"
      >
        <button
          type="button"
        >
          X
        </button>
      </div>
    </div>
    <div
      class="wrapper"
    >
      Total: 
      5000.25
      $
    </div>
  </div>
</div>
`;
